### PR TITLE
Migrate Movement 2 (BatchToSpace-2, DepthToSpace-1, Roll-7, ShuffleCha…

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Create template plugin reference tests for target operation: BatchToSpace-2, DepthToSpace-1, Roll-7, ShuffleChannels-1, SpaceToBatch-2, SpaceToDepth-1*
 - *Remove ngraph backend files*
 - *Update CMakeLists.txt*
 - *Update skip_tests_config.cpp:*
 - *R"(.*ReferenceBatchToSpaceLayerTest.*_input_with_shape_with_crop_4x1x2x3)",*
 - *R"(.*ReferenceBatchToSpaceLayerTest.*_input_with_shape_with_crop_4x1x2x3_1)",*
 - *R"(.*ReferenceSpaceToBatchLayerTest.*_space_to_batch_5D)",*
 - *R"(.*ReferenceSpaceToBatchLayerTest.*_space_to_batch_4x4)",*
 
### Tickets:
 - *64006*
 - *63953*
 - *64113*
 - *63988*
 - *64050*
 - *63993*

### CODE REVIEW

**Comment by steve-y** [comment]
@iefode @ilyachur The failed cases are described as below, could you check them please?

BatchToSpace has the following error so they are added skip_tests_config.cpp
```
Failed case 1:

[ RUN      ] smoke_BatchToSpace_With_Hardcoded_Refs/ReferenceBatchToSpaceLayerTest.CompareWithRefs/dType=i8_dShape={4, 1, 2, 3}_bsType=i64_bsShape={4}_cbType=i64_cbShape={4}_ceType=i64_ceShape={4}_eType=i8_eShape={1, 1, 4, 4}_input_with_shape_with_crop_4x1x2x3
Access PluginCache ov core. OV Core use count: 1

Available devices for TEMPLATE:
../docs/template_plugin/tests/functional/op_reference/base_reference_test.cpp:80: Failure
Expected equality of these values:
  refBlob.get_byte_size()
    Which is: 16
  outBlob.get_byte_size()
    Which is: 24
```
```
Failed case 2:

[ RUN      ] smoke_BatchToSpace_With_Hardcoded_Refs/ReferenceBatchToSpaceLayerTest.CompareWithRefs/dType=i8_dShape={4, 1, 2, 3}_bsType=i64_bsShape={4}_cbType=i64_cbShape={4}_ceType=i64_ceShape={4}_eType=i8_eShape={1, 1, 4, 4}_input_with_shape_with_crop_4x1x2x3_1
Access PluginCache ov core. OV Core use count: 0
Created ov core.
Available devices for TEMPLATE:

../docs/template_plugin/tests/functional/op_reference/base_reference_test.cpp:80: Failure
Expected equality of these values:
  refBlob.get_byte_size()
    Which is: 16
  outBlob.get_byte_size()
    Which is: 8
```

SpaceToBatch has the following error so they are added skip_tests_config.cpp
```
Failed case 1:

[ RUN      ] smoke_SpaceToBatch_With_Hardcoded_Refs/ReferenceSpaceToBatchLayerTest.CompareWithRefs/dType=i8_dShape={1, 1, 3, 2, 1}_bsType=i64_bsShape={5}_pbType=i64_pbShape={5}_peType=i64_peShape={5}_eType=i8_eShape={12, 1, 2, 1, 2}_space_to_batch_5D
Access PluginCache ov core. OV Core use count: 1
Available devices for TEMPLATE:

unknown file: Failure
C++ exception with description "Check '(pads_begin_val.at(idx) + data_shape.at(idx) + pads_end_val.at(idx)) % block_val.at(idx) == 0' failed at core/src/op/space_to_batch.cpp:97:
While validating node 'v1::SpaceToBatch SpaceToBatch_1940 (Parameter_1936[0]:i8{1,1,3,2,1}, Constant_1937[0]:i64{5}, Constant_1938[0]:i64{5}, Constant_1939[0]:i64{5}) -> (dynamic?)' with friendly_name 'SpaceToBatch_1940':
The dimension on position: 2 equal to: 5 must be a multiple of block_values[i]: 3
" thrown in SetUp().
```
```
Failed case 2:

[ RUN      ] smoke_SpaceToBatch_With_Hardcoded_Refs/ReferenceSpaceToBatchLayerTest.CompareWithRefs/dType=i8_dShape={1, 1, 4, 4}_bsType=i64_bsShape={4}_pbType=i64_pbShape={4}_peType=i64_peShape={4}_eType=i8_eShape={1, 1, 5, 4}_space_to_batch_4x4
Access PluginCache ov core. OV Core use count: 0
Created ov core.
Available devices for TEMPLATE:

../docs/template_plugin/tests/functional/op_reference/base_reference_test.cpp:80: Failure
Expected equality of these values:
  refBlob.get_byte_size()
    Which is: 20
  outBlob.get_byte_size()
    Which is: 24
```

But the SpaceToBatch failed case 2, R"(.ReferenceSpaceToBatchLayerTest._space_to_batch_5D)" in skip_tests_config.cpp does not work, space_to_batch_5D part is commented out in space_to_batch.cpp for now.

File: docs/template_plugin/tests/functional/op_reference/batch_to_space.cpp [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        generateBatchToSpaceParams<element::Type_t::i16>(),
+        generateBatchToSpaceParams<element::Type_t::i32>(),
+        generateBatchToSpaceParams<element::Type_t::i64>(),
+        generateBatchToSpaceParams<element::Type_t::u8>(),
+        generateBatchToSpaceParams<element::Type_t::u16>(),
+        generateBatchToSpaceParams<element::Type_t::u32>(),
+        generateBatchToSpaceParams<element::Type_t::u64>(),
+        generateBatchToSpaceParams<element::Type_t::f16>(),
+        generateBatchToSpaceParams<element::Type_t::f32>(),
+        generateBatchToSpaceParams<element::Type_t::f64>(),
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/batch_to_space.cpp:209]: In general we don't have any limitations for other data types, like `bf16`, `u4` and etc. in `BatchToSpace::evaluate()` method. Could we extend the set of supported data types?

File: docs/template_plugin/tests/functional/op_reference/depth_to_space.cpp [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        generateDepthToSpaceParams<element::Type_t::i32>(),
+        generateDepthToSpaceParams<element::Type_t::i64>(),
+        generateDepthToSpaceParams<element::Type_t::u8>(),
+        generateDepthToSpaceParams<element::Type_t::u16>(),
+        generateDepthToSpaceParams<element::Type_t::u32>(),
+        generateDepthToSpaceParams<element::Type_t::u64>(),
+        generateDepthToSpaceParams<element::Type_t::bf16>(),
+        generateDepthToSpaceParams<element::Type_t::f16>(),
+        generateDepthToSpaceParams<element::Type_t::f32>(),
+        generateDepthToSpaceParams<element::Type_t::f64>(),
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/depth_to_space.cpp:146]: The same comment as above.

File: docs/template_plugin/tests/functional/op_reference/shuffle_channels.cpp [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    const std::vector<std::vector<ShuffleChannelsParams>> generatedParams {
+        generateParams<element::Type_t::i8>(),
+        generateParams<element::Type_t::i16>(),
+        generateParams<element::Type_t::i32>(),
+        generateParams<element::Type_t::i64>(),
+        generateParams<element::Type_t::u8>(),
+        generateParams<element::Type_t::u16>(),
+        generateParams<element::Type_t::u32>(),
+        generateParams<element::Type_t::u64>(),
+        generateParams<element::Type_t::bf16>(),
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/shuffle_channels.cpp:194]: The same

**Comment by steve-y** [comment]
@Ilya I updated to use opsetN, and add type i4, u4, and bf16. Adding bf16 has no issue, but adding i4 and u4 have the following issues. And I'm not sure if it needs more debugging or creating tickets for them. Summarize them as follows;
> batch_to_space.cpp, roll.cpp, space_to_batch.cpp are like generating 0 when the result values are over 7. So when making them 0 makes the test passed. So they are included to skip_tests_config.cpp.
> depth_to_space.cpp, shuffle_channels.cpp generate segmentation faults when they generate and register test cases. So the test case are not added.
> space_to_depth.cpp generate the core dumped after run. So the test case are not added.

Log Summary:

1. batch_to_space.cpp
`../inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/base/layer_test_utils.hpp:118 Relative comparison of values expected: 8 and actual: 0 at index 0 with threshold 0.01 failed" thrown in the test body.`

2. roll.cpp
`../inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/base/layer_test_utils.hpp:118 Relative comparison of values expected: 8 and actual: 0 at index 0 with threshold 0.01 failed" thrown in the test body.`

3. space_to_batch.cpp
`C++ exception with description "partial tensor write not supported" thrown in the test body.`
`corrupted size vs. prev_size
Aborted (core dumped)`

4. depth_to_space.cpp
`Segmentation fault (core dumped)`

5. shuffle_channels.cpp
`Segmentation fault (core dumped)`

6. space_to_depth.cpp
`free(): invalid size
Aborted (core dumped)`

**Comment by ilyachur** [comment]
> @ilya I updated to use opsetN, and add type i4, u4, and bf16. Adding bf16 has no issue, but adding i4 and u4 have the following issues. Summarize them as follows;
> 
>     1. batch_to_space.cpp, roll.cpp are like generating 0 when the result values are over 7. So when making them 0 makes the test passed. And they seem to be accuracy drop in i4 and u4. So they are added to skip_tests_config.cpp. And they need to create tickets and follow up.
>        `../inference-engine/tests/functional/shared_test_classes/include/shared_test_classes/base/layer_test_utils.hpp:118 Relative comparison of values expected: 8 and actual: 0 at index 0 with threshold 0.01 failed" thrown in the test body.`
> 
>     2. depth_to_space.cpp, shuffle_channels.cpp generate segmentation faults when they generate and register test cases, and they could not be skipped with skip_tests_config.cpp. So the test cases are not added. They seem similar to Constant CPU case in https://jira.devtools.intel.com/browse/CVS-69642. So they need to create tickets and follow up.
>        `Segmentation fault (core dumped)`
> 
>     3. space_to_batch.cpp, space_to_depth.cpp generate the core dumped after run. And they seem to generate errors in their destructors or free phase, and they could not be skipped with skip_tests_config.cpp. So the test cases are not added. And they need to create tickets and follow up.
> 
> 
> > space_to_batch.cpp
> > `C++ exception with description "partial tensor write not supported" thrown in the test body.`
> > `corrupted size vs. prev_size Aborted (core dumped)`
> > space_to_depth.cpp
> > `free(): invalid size Aborted (core dumped)`

Ok, thank you for the investigation.
I propose to disable failed tests and prohibit unsupported element types and create tickets to know about these limitations.

**Comment by steve-y** [comment]
Remove the test i4 and u4 from BatchToSpace and Roll, because they have similar issue on Windows machines, like Constant (https://jira.devtools.intel.com/browse/CVS-69642) and create a ticket for this (https://jira.devtools.intel.com/browse/CVS-70598).

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
Please resolve merge conflicts

**Pull request merged**